mlir/lib/Dialect/SCF/Utils/Utils.cpp

```diff
@@ -490,7 +494,8 @@ LogicalResult mlir::loopUnrollByFactor(
       annotateFn, iterArgs, yieldedValues);
   // Promote the loop body up if this has turned into a single iteration loop.
   (void)forOp.promoteIfSingleIteration(rewriter);
-  return success();
+  resultLoops.mainLoopOp = forOp;
```

Comment:
if the promotion succeeded the forOp is delete. Same above.

Reply:
Good catch! In this case looks like we cannot rely on when the resulting main loop is null for success or failure, perhaps `resultLoops` should be returned by a byref parameter?

Reply:
best would be to use `FailureOr<UnrolledLoopInfo>`